staging/cse/windows/nvidiagpudriverfunc.ps1

```diff
@@ -0,0 +1,337 @@
+    $env:Path = "$PSScriptRoot;$env:Path"
+  }
+
+  try {
+    $FatalError = @()
+
+    $LogFolder = "$PSScriptRoot\.."
+
+    $Reboot = @{
+      Needed = $false
+    }
+
+    Write-Host "Attempting to install Nvidia driver..."
+
+    # Get the SetupTarget based on the input
+    $Setup = Get-Setup
+    $SetupTarget = $Setup.Target
+    $Reboot.Needed = $Setup.RebootNeeded
+    Write-Host "Setup complete"
+
+    Add-DriverCertificate $Setup.CertificateUrl
+    Write-Host "Certificate in store"
+
+    Write-Host "Installing $SetupTarget ..."
+    try {
+      $InstallLogFolder = "$LogFolder\NvidiaInstallLog"
+      $Arguments = "-s -n -log:$InstallLogFolder -loglevel:6"
+    
+      $p = Start-Process -FilePath $SetupTarget -ArgumentList $Arguments -PassThru
+
+      $Timeout = 10 * 60 # in seconds. 10 minutes for timeout of the installation
+      Wait-Process -InputObject $p -Timeout $Timeout -ErrorAction Stop
+    
+      # check if installation was successful
+      if ($p.ExitCode -eq 0 -or $p.ExitCode -eq 1) {
+        # 1 is issued when reboot is required after success
+        Write-Host "GPU Driver Installation Success. Code: $($p.ExitCode)"
+      }
+      else {
+        Write-Host "GPU Driver Installation Failed! Code: $($p.ExitCode)"
+      }
+
+      if ($Reboot.Needed -or $p.ExitCode -eq 1) {
+        Write-Host "Reboot is needed for this GPU Driver..."
+        $result.RebootNeeded = $true
+      }
+      return $result
+    }
+    catch [System.TimeoutException] {
+      Write-Host "Timeout $Timeout s exceeded. Stopping the installation process. Reboot for another attempt."
+      Stop-Process -InputObject $p
+    }
+    catch {
+      $Message = $_.ToString()
+      Write-Host "Exception: $Message" # the status file may get over-written when the agent re-attempts this step
+      throw
+    }
+    
+  }
+  catch [NotSupportedException] {
+    $Message = $_.ToString()
+    Write-Host $Message
+    exit 52
+  }
+  catch {
+    $FatalError += $_
+    Write-Host -ForegroundColor Red ($_ | Out-String)
+    exit $FatalError.Count
+  }
+}
+
+function Get-Setup {
+  [OutputType([hashtable])]
+    
+  # Choose driver and specific properties
+  $Driver = Select-Driver
+    
+  if ($Driver.Url -eq $null -or $Driver.CertificateUrl -eq $null) {
+    $Message = "DriverURL or DriverCertificateURL are not properly specified."
+    throw $Message
+  }
+    
+  $Setup = @{
+    RebootNeeded   = $Driver.RebootNeeded
+    CertificateUrl = $Driver.CertificateUrl
+  }
+        
+  $InstallFolder = "$PSScriptRoot\.."
+    
+  $source = $Driver.Url
+  $dest = "$InstallFolder\$($Driver.InstallExe)"
+
+  Get-DriverFile $source $dest
+
+  $Setup.Target = "$InstallFolder\$($Driver.InstallExe)"
+
+  return $Setup
+}
+
+function Select-Driver {
+  [OutputType([hashtable])]
+
+  # Set some default values
+  $Driver = @{
+    # Hard coding the FwLink for resources.json
+    ResourceUrl  = "https://go.microsoft.com/fwlink/?linkid=2181101"
```

Comment:
Does this work in egress test? Some customers only allow those public FQDNs in their firewall. https://learn.microsoft.com/en-us/azure/aks/outbound-rules-control-egress